### PR TITLE
Feature/cascading config

### PR DESCRIPTION
**Pull request by dLabsMarijnB**
Feature that allows for cascading configuration.
The order it will load the config is:
- splunk_defaults.yml
- /tmp/defaults/default.yml
- Download from the ENV var specified by default to be: SPLUNK_DEFAULTS_URL

Afterwards in the dynamic inventory it will check if /.dockerenv exists, if so, add localhost and try to pull from the url (which should be specified in one of the above configurations) and put this in the hostvars

For example:
```
config:
    host:
        url: https://<server>/{hostname}.yml
        headers:
            TOKEN: <token>
        verify: True
```

Using these parameters you can allow for authenticated, cert verified or not, download of a dynamic link.

For example:
This would allow to specify a deploymentserver in de defaults.
But for the deploymentserver, override it to null, thus all but the deploymentserver will get this setting.

### CODE REVIEW

**Comment by dLabsMarijnB** [comment]
Seems the identing of the above yaml got messed up.
`config:
    host:
        url: https://<server>/{hostname}.yml
        headers:
            TOKEN: <token>
        verify: True`

File: inventory/splunk_defaults.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        url:
+        headers:
+        verify: True
+
+splunk:
+    opt: &opt "/opt"
+    home: &home !!python/object/apply:string.join [[*opt, "/splunk"],'']
+    user: "splunk"
+    group: "splunk"
+    exec: !!python/object/apply:string.join [[*home, "/bin/splunk"], '']
```
Comment (lephino) [inline inventory/splunk_defaults.yml]: Won't this break on windows?

File: inventory/environ.py [hunk 2]
```diff
@@ -126,150 +132,114 @@ def getSplunkApps(inv):
...
-    vars_scope["splunk"]["idxc"]["search_factor"] = getValueFromDEFAULTS("splunk.idxc.search_factor", casting=int) or 3
-    vars_scope["splunk"]["idxc"]["replication_factor"] = getValueFromDEFAULTS("splunk.idxc.replication_factor", casting=int) or 3
-    vars_scope["splunk"]["idxc"]["replication_port"] = getValueFromDEFAULTS("splunk.idxc.replication_port", casting=int) or 4001
-    vars_scope["splunk"]["enable_service"] = os.environ.get('SPLUNK_ENABLE_SERVICE') or False
-    vars_scope["splunk"]["smartstore"] = getValueFromDEFAULTS("splunk.smartstore")
-
-    vars_scope["splunk"]["app_paths"] = {}
-    if PLATFORM == "linux":
-        vars_scope["splunk"]["group"] = getValueFromDEFAULTS("splunk.group") or "splunk"
-        vars_scope["splunk"]["exec"] = getValueFromDEFAULTS(
```
Comment (lephino) [inline inventory/environ.py]: The main reasons for these being defined in here is to have python handle the pathing for different OS's.  My only real concern is windows support.

**Comment by dLabsMarijnB** [comment]
Maybe the simplest and cleanest solution would be to put the platform variable inside of the splunk_defaults.yml file.
e.g.
splunk_defaults_linux.yml & splunk_defaults_windows.yml
Keeps it simple and maintance a good overview of what goes where.

Thoughts?

**Comment by dLabsMarijnB** [comment]
There we go, now all the urls can be platform specific as well.

**Pull request merged**